### ISSUE TITLE
Random crashes using session replay

### ISSUE THREAD

**Issue by paulocesar-ufu** [DataDog/dd-sdk-android#1795]
Hello folks!

After activating the session replay feature in our app, we noticed some random crashes.
Fortunately, all the stack traces are the same, leading me to believe that the bug is at a single point:

```
Fatal Exception: java.util.ConcurrentModificationException

java.util.HashMap$HashIterator.nextNode (HashMap.java:1441)
java.util.HashMap$KeyIterator.next (HashMap.java:1465)
com.datadog.android.sessionreplay.internal.recorder.base64.BitmapPool.get (BitmapPool.kt:98)
com.datadog.android.sessionreplay.internal.recorder.base64.BitmapPool.getBitmapByProperties$dd_sdk_android_session_replay_release (BitmapPool.kt:108)
com.datadog.android.sessionreplay.internal.utils.DrawableUtils.getBitmapBySize (DrawableUtils.kt:188)
com.datadog.android.sessionreplay.internal.utils.DrawableUtils.createScaledBitmap (DrawableUtils.kt:146)
com.datadog.android.sessionreplay.internal.utils.DrawableUtils.createBitmapOfApproxSizeFromDrawable$lambda$0 (DrawableUtils.kt:57)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
java.lang.Thread.run (Thread.java:920)
```

**We are using the lastest dd-sdk-android v2.3.0**
It's worth noting that this issue occurs on different Android versions (12, 13, 14) and on different device models.

Thanks in advance!

**Comment by mariusc83** [comment]
Hi @paulocesar-ufu, thank you for reporting this, we might indeed have a bug in there. We will analyze this internally and come back with a conclusion/resolution in short time.

**Comment by paulocesar-ufu** [comment]
Thank you @mariusc83, let me know if you need any more information.

**Comment by mariusc83** [comment]
@paulocesar-ufu we analyzed this issue on our end and we can assure you that it will be fixed in the next version. We decided not to have a hot fix for this as this is still relatively small based on our current metrics. Let us know if this works for you. Thanks !!

**Comment by paulocesar-ufu** [comment]
That sounds good @mariusc83.
Indeed, this crash affects less than 1% of our users.
For now, I'll reduce the sample rate until the new version is released.

Thanks!

**Comment by mariusc83** [comment]
@paulocesar-ufu this was fixed in https://github.com/DataDog/dd-sdk-android/pull/1798. It will be available in the next release.

**Comment by 0xnm** [comment]
Hello @paulocesar-ufu! The fix is available in the release `2.5.0`.
